### PR TITLE
[Modular] Chameleon Mutation Fix

### PR DESCRIPTION
**Pull request by Nerev4r**
## About The Pull Request
It accidentally made you slightly more invisible than 255 if you clicked on stuff. Now it don't.

## How This Contributes To The Skyrat Roleplay Experience
Allows me to stop facing existential dread about how I occasionally become slightly transparent when I'm using this mutation.

## Changelog
:cl:
fix: Fixes a minor bug with chameleon mutation and transparency.
/:cl:

### CODE REVIEW

File: code/datums/mutations/chameleon.dm [hunk 1]
```diff
@@ -7,26 +7,26 @@
...
-	action_icon_state = "ninja_cloak"
+	action_icon_state = "ninja_cloak" //SKYRAT EDIT END
 
 /datum/mutation/human/chameleon/on_acquiring(mob/living/carbon/human/owner)
 	if(..())
 		return
-	if(HAS_TRAIT(owner,TRAIT_CHAMELEON_SKIN))
-		return
+	if(HAS_TRAIT(owner,TRAIT_CHAMELEON_SKIN)) //SKYRAT EDIT BEGIN
+		return //SKYRAT EDIT END
```
Comment (GoldenAlpharex) [inline code/datums/mutations/chameleon.dm]:
```suggestion
	/// SKYRAT EDIT BEGIN
	if(HAS_TRAIT(owner,TRAIT_CHAMELEON_SKIN))
		return 
	/// SKYRAT EDIT END
```

File: code/datums/mutations/chameleon.dm [hunk 2]
```diff
@@ -7,26 +7,26 @@
...
 	owner.alpha = CHAMELEON_MUTATION_DEFAULT_TRANSPARENCY
 	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, .proc/on_move)
 	RegisterSignal(owner, COMSIG_HUMAN_EARLY_UNARMED_ATTACK, .proc/on_attack_hand)
 
-/obj/effect/proc_holder/spell/self/chameleon_skin_activate/cast(list/targets, mob/user = usr)
+/obj/effect/proc_holder/spell/self/chameleon_skin_activate/cast(list/targets, mob/user = usr) //SKYRAT EDIT BEGIN
```
Comment (GoldenAlpharex) [inline code/datums/mutations/chameleon.dm]: If those two procs are self-contained, then they should be moved to modular. Man, I really wasn't awake when I reviewed that last PR, my apologies.

File: code/datums/mutations/chameleon.dm [hunk 3]
```diff
@@ -42,26 +42,29 @@
 
 	REMOVE_TRAIT(user, TRAIT_CHAMELEON_SKIN, GENETIC_MUTATION)
 	user.alpha = 255
-	to_chat(user, text("Your skin shifts as it shimmers back into its original colors."))
+	to_chat(user, text("Your skin shifts as it shimmers back into its original colors.")) //SKYRAT EDIT END
 
 /datum/mutation/human/chameleon/on_life(delta_time, times_fired)
```
Comment (GoldenAlpharex) [inline code/datums/mutations/chameleon.dm:26]: Same as previous comment

File: code/datums/mutations/chameleon.dm [hunk 4]
```diff
@@ -42,26 +42,29 @@
...
-	if(HAS_TRAIT(owner, TRAIT_CHAMELEON_SKIN))
-		owner.alpha = max(owner.alpha - (12.5 * delta_time), 0)
+	if(HAS_TRAIT(owner, TRAIT_CHAMELEON_SKIN)) //SKYRAT EDIT BEGIN
+		owner.alpha = max(owner.alpha - (12.5 * delta_time), 0) //SKYRAT EDIT END
 
 /datum/mutation/human/chameleon/proc/on_move()
 	SIGNAL_HANDLER
 
-	if(HAS_TRAIT(owner, TRAIT_CHAMELEON_SKIN))
+	if(HAS_TRAIT(owner, TRAIT_CHAMELEON_SKIN)) //SKYRAT EDIT BEGIN
```
Comment (GoldenAlpharex) [inline code/datums/mutations/chameleon.dm]:
```suggestion
	/// SKYRAT EDIT BEGIN
	if(HAS_TRAIT(owner, TRAIT_CHAMELEON_SKIN))
```

File: code/datums/mutations/chameleon.dm [hunk 5]
```diff
@@ -42,26 +42,29 @@
...
 		owner.alpha = CHAMELEON_MUTATION_DEFAULT_TRANSPARENCY
 	else
-		owner.alpha = 255
+		owner.alpha = 255 //SKYRAT EDIT END
```
Comment (GoldenAlpharex) [inline code/datums/mutations/chameleon.dm]:
```suggestion
		owner.alpha = 255
		/// SKYRAT EDIT END
```

File: code/datums/mutations/chameleon.dm [hunk 6]
```diff
@@ -42,26 +42,29 @@
...
 /datum/mutation/human/chameleon/proc/on_attack_hand(atom/target, proximity)
 	SIGNAL_HANDLER
 
 	if(!proximity) //stops tk from breaking chameleon
 		return
-	owner.alpha = CHAMELEON_MUTATION_DEFAULT_TRANSPARENCY
+	if(HAS_TRAIT(owner, TRAIT_CHAMELEON_SKIN)) //SKYRAT EDIT BEGIN
+		owner.alpha = CHAMELEON_MUTATION_DEFAULT_TRANSPARENCY
+	else
+		owner.alpha = 255 //SKYRAT EDIT END
```
Comment (GoldenAlpharex) [inline code/datums/mutations/chameleon.dm]:
```suggestion
	/// SKYRAT EDIT BEGIN
	if(HAS_TRAIT(owner, TRAIT_CHAMELEON_SKIN))
		owner.alpha = CHAMELEON_MUTATION_DEFAULT_TRANSPARENCY
	else
		owner.alpha = 255
	/// SKYRAT EDIT END
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
I'm sorry, I didn't do my job at reviewing your previous PR properly, so there might be a bit more work involved in this PR.

File: code/datums/mutations/chameleon.dm [hunk 7]
```diff
@@ -7,61 +7,53 @@
...
-
-	REMOVE_TRAIT(user, TRAIT_CHAMELEON_SKIN, GENETIC_MUTATION)
-	user.alpha = 255
-	to_chat(user, text("Your skin shifts as it shimmers back into its original colors."))
-
 /datum/mutation/human/chameleon/on_life(delta_time, times_fired)
-	if(HAS_TRAIT(owner, TRAIT_CHAMELEON_SKIN))
-		owner.alpha = max(owner.alpha - (12.5 * delta_time), 0)
+	if(HAS_TRAIT(owner, TRAIT_CHAMELEON_SKIN)) //SKYRAT EDIT BEGIN
+		owner.alpha = max(owner.alpha - (12.5 * delta_time), 0) //SKYRAT EDIT END
```
Comment (GoldenAlpharex) [inline code/datums/mutations/chameleon.dm]:
```suggestion
	/// SKYRAT EDIT BEGIN
	if(HAS_TRAIT(owner, TRAIT_CHAMELEON_SKIN))
		owner.alpha = max(owner.alpha - (12.5 * delta_time), 0)
	/// SKYRAT EDIT END
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
One last nitpick.

**Comment by Nerev4r** [comment]
;w;

File: code/datums/mutations/chameleon.dm [hunk 8]
```diff
@@ -7,61 +7,55 @@
...
+	instability = 35 //SKYRAT EDIT BEGIN
 	power = /obj/effect/proc_holder/spell/self/chameleon_skin_activate
 
 /obj/effect/proc_holder/spell/self/chameleon_skin_activate
 	name = "Activate Chameleon Skin"
 	desc = "The chromatophores in your skin adjust to your surroundings, as long as you stay still."
 	clothes_req = FALSE
 	action_icon = 'icons/mob/actions/actions_minor_antag.dmi'
-	action_icon_state = "ninja_cloak"
+	action_icon_state = "ninja_cloak" //SKYRAT EDIT END
```
Comment (GoldenAlpharex) [inline code/datums/mutations/chameleon.dm]: That needs to be modularized too ;)

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: code/datums/mutations/chameleon.dm [hunk 9]
```diff
@@ -7,61 +7,48 @@
...
-	power = /obj/effect/proc_holder/spell/self/chameleon_skin_activate
-
-/obj/effect/proc_holder/spell/self/chameleon_skin_activate
-	name = "Activate Chameleon Skin"
-	desc = "The chromatophores in your skin adjust to your surroundings, as long as you stay still."
-	clothes_req = FALSE
-	action_icon = 'icons/mob/actions/actions_minor_antag.dmi'
-	action_icon_state = "ninja_cloak"
+	instability = 35 //SKYRAT EDIT BEGIN
+	power = /obj/effect/proc_holder/spell/self/chameleon_skin_activate //SKYRAT EDIT END
```
Comment (GoldenAlpharex) [inline code/datums/mutations/chameleon.dm]: I'll commit this myself, it'll be faster.
```suggestion
	 /// SKYRAT EDIT BEGIN
	instability = 35
	power = /obj/effect/proc_holder/spell/self/chameleon_skin_activate
	/// SKYRAT EDIT END
```

**Review by GoldenAlpharex — APPROVED** [review]
Good work. We got there at the end.

**Comment by GoldenAlpharex** [comment]
At long last.

**Pull request merged**